# $Id$
#
# @TEST-EXEC:  hilti-build %INPUT -o a.out
# @TEST-EXEC:  ./a.out >output 2>&1
# @TEST-EXEC:  btest-diff output

module Main

import Hilti

type myException = exception

void run() {

     call Hilti::print ("A")

     try {

       local ref<myException> e
       e = new myException
       exception.throw e

       call Hilti::print ("Cannot be reached")
     }

     catch myException {
       call Hilti::print ("Caught it!")
     }

     call Hilti::print ("C")
     return.void

}

